rpcs3/Emu/RSX/GL/GLProgramBuffer.h

```diff
@@ -134,12 +134,12 @@ struct GLProgramBuffer : public program_state_cache<GLTraits>
 	}
 
 	template <typename... Args>
-	void add_pipeline_entry(RSXVertexProgram &vp, RSXFragmentProgram &fp, void* &props, Args&& ...args)
+	void add_pipeline_entry(const RSXVertexProgram& vp, const RSXFragmentProgram& fp, void* &props, Args&& ...args)
 	{
 		get_graphics_pipeline(vp, fp, props, false, false, std::forward<Args>(args)...);
 	}
 
-	void preload_programs(RSXVertexProgram &vp, RSXFragmentProgram &fp)
+	void preload_programs(const RSXVertexProgram& vp, const RSXFragmentProgram& fp)
```

Comment:
I'm not sure about this one, though if it works then it's ok for now, we can always roll back. There is no architectural/design reason to not modify the program record when preloading the program to optimize work later down the line.

Reply:
it's always bad to have non-const ref without reason. So I'd rather have const as long as possible.